flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -606,7 +608,12 @@ private void releaseEmptyTaskManager(ResourceID resourceId) {
 
 		final TaskManagerLocation taskManagerLocation;
 		try {
-			taskManagerLocation = TaskManagerLocation.fromUnresolvedLocation(unresolvedTaskManagerLocation);
+			final boolean retrieveTaskManagerHostName = this.jobMasterConfiguration.getConfiguration()
+					.getBoolean(JobManagerOptions.RETRIEVE_TASK_MANAGER_HOSTNAME);
+			if (!retrieveTaskManagerHostName) {
+				log.info("JobManager would not retrieve TaskManager's canonical hostname due to configuration.");
```

Comment:
I don't think it is a good idea to log this message every time we register a taskmanager.

On startup, we are logging all configuration parameters. I think that is enough in this case OR we log on debug in the constructor when looking up the config key?

Reply:
Yes this is a little verbose to log on every registration, and since we have already printed all the configuration parameters during startup, this logging is removed in the new commit : )